fdbserver/MasterProxyServer.actor.cpp

```diff
@@ -618,6 +628,14 @@ ACTOR Future<Void> commitBatch(
 	state vector<vector<int>> transactionResolverMap = std::move( requests.transactionResolverMap );
 	state Future<Void> releaseFuture = releaseResolvingAfter(self, releaseDelay, localBatchNumber);
 
+	if(localBatchNumber-self->latestLocalCommitBatchLogging.get()>SERVER_KNOBS->RESET_RESOLVER_BATCHES && now()-self->lastResolverReset>SERVER_KNOBS->RESET_RESOLVER_DELAY) {
+		TraceEvent(SevWarnAlways, "ResetResolverNetwork").detail("CurrentBatch", localBatchNumber).detail("InProcessBatch", self->latestLocalCommitBatchLogging.get());
+		for (int r = 0; r<self->resolvers.size(); r++) {
+			FlowTransport::transport().resetConnection(self->resolvers[r].address());
+		}
+		self->lastResolverReset=now();
+	}
```

Comment:
In a scenario where the round-trip latency between the proxy and the resolver hits 1 second but everything else behaves normally, we're going to end up with a number of outstanding batches to the resolver roughly equal to the number of commits we are attempting per second? 

If so, then that means this logic only helps us if our commit rate exceeds the chosen batch limit divided by the latency. It's better than before but probably misses a lot of cases.

The master request is a bit different in that we only have one outstanding at a time. In that case, we would trigger the behavior whenever the latency is larger than 1/commit rate, since we would fail to keep up and the batch count would grow. It still has blind spots if the commit rate is very low, but in general the sensitivity is much higher.

Reply:
This is a good point. I guess this PR does not solve all potential issues, but the most emergent ones. 

Would you like to formulate the general problem: if a proxy has slow network with master or resolver role, how to detect it? 
(Once we know, we can reset network.)

Reply:
A more general version of the problem that this is addressing is connections getting into a high latency state that is fixed by a reset.